### PR TITLE
Back port branch-1.0 to master

### PR DESCRIPTION
**Pull request by apc999**
Since branch-1.0 has a number of bug/doc fixes, this PR aims to bring those fixes into master

### CODE REVIEW

**Comment by apc999** [comment]
Jenkins, test this please

File: core/server/src/test/java/alluxio/worker/block/BlockWorkerTest.java [hunk 1]
```diff
@@ -106,7 +107,7 @@ public void before() throws IOException {
   }
 
   /**
-   * Resets the worker context.
+   * Reset the worker context and close the data server on clean up.
    */
   @After
   public void after() {
```
Comment (calvinjia) [inline core/server/src/test/java/alluxio/worker/block/BlockWorkerTest.java]: Did we lose a line due to merge conflict? (data server close)

File: docs/_config.yml [hunk 2]
```diff
@@ -5,8 +5,8 @@ kramdown:
   syntax_highlighter: rouge
 
 # These allow the documentation to be updated with new releases of Alluxio.
-ALLUXIO_RELEASED_VERSION: 1.0.0
-ALLUXIO_MASTER_VERSION_SHORT: 1.1.0-SNAPSHOT
+ALLUXIO_RELEASED_VERSION: 1.0.1
+ALLUXIO_MASTER_VERSION_SHORT: 1.0.2-SNAPSHOT
```
Comment (calvinjia) [inline docs/_config.yml]: Master version should still be `1.1.0-SNAPSHOT` right?

**Comment by calvinjia** [comment]
@apc999 Two comments otherwise LGTM

File: docs/_config.yml (hunk 2 above)
Comment (apc999) [inline docs/_config.yml]: done

File: core/server/src/test/java/alluxio/worker/block/BlockWorkerTest.java (hunk 1 above)
Comment (apc999) [inline core/server/src/test/java/alluxio/worker/block/BlockWorkerTest.java]: Good catch. Fixed

**Comment by apc999** [comment]
@calvinjia  Comments addressed. PTAL
Thanks

File: core/server/src/test/java/alluxio/worker/block/BlockWorkerTest.java [hunk 3]
```diff
@@ -106,10 +108,11 @@ public void before() throws IOException {
   }
 
   /**
-   * Resets the worker context.
+   * Resets the worker context and close the data server on clean up.
    */
   @After
-  public void after() {
+  public void after() throws IOException {
+    ((DataServer) Whitebox.getInternalState(mBlockWorker, "mDataServer")).close();
```
Comment (calvinjia) [inline core/server/src/test/java/alluxio/worker/block/BlockWorkerTest.java]: Oh my bad, this should be removed (as BlockWorker does not hold the data server anymore)

File: core/server/src/test/java/alluxio/worker/block/BlockWorkerTest.java [hunk 4]
```diff
@@ -106,10 +108,11 @@ public void before() throws IOException {
   }
 
   /**
-   * Resets the worker context.
+   * Resets the worker context and close the data server on clean up.
```
Comment (calvinjia) [inline core/server/src/test/java/alluxio/worker/block/BlockWorkerTest.java]: This comment is the line which should have been changed to just reset the worker context.

**Comment by calvinjia** [comment]
@apc999 Looks like I was wrong about the merge conflict, we should modify the comment instead of the logic.

**Comment by apc999** [comment]
@calvinjia  PTAL, thanks

**Comment by calvinjia** [comment]
LGTM

File: core/client/src/main/java/alluxio/client/file/FileInStream.java [hunk 5]
```diff
@@ -69,6 +70,9 @@
   /** Constant error message for block ID not cached. */
   private static final String BLOCK_ID_NOT_CACHED =
       "The block with ID {} could not be cached into Alluxio storage.";
+  /** Error message for cache collision. */
+  private static final String BLOCK_ID_EXISTS_SO_NOT_CACHED =
```
Comment (jsimsa) [inline core/client/src/main/java/alluxio/client/file/FileInStream.java]: nit: I think this should be called BLOCK_ID_ALREADY_EXISTS

**Comment by jsimsa** [comment]
just one minor nit that can be addressed in a separate PR

**Pull request merged**